packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2594,7 +2738,16 @@ export function writeInitialResources(
   destination: Destination,
   resources: Resources,
   responseState: ResponseState,
+  willFlushAllSegments: boolean,
 ): boolean {
+  if (
+    enableFizzExternalRuntime &&
+    !willFlushAllSegments &&
+    responseState.externalRuntimeConfig
```

Comment:
i might leave a comment here explaining why this is the case since it may not be readily apparent to an unfamiliar reader